### ISSUE TITLE
Preact-Vite: $RefreshReg$ is not defined 

### ISSUE THREAD

**Issue by theKashey** [callstack/linaria#1195]
This is just a report of an error that might occur with linaria, but the one (probably) not caused by linaria, and not to be fixed by it.

## Environment
Enviroment is [@preact/preset-vite](https://www.npmjs.com/package/@preact/preset-vite) configured by storybook. Root cause - `@prefresh` enabled.

## Description

The issue was first detected in Storybook 7 `preact-vite` environment
```
[vite] Internal server error: $RefreshReg$ is not defined in
 at .../node_modules/.pnpm/@linaria+babel-preset@4.3.3/node_modules/@linaria/babel-preset/lib/module.js:348:15

```

Problem only occurs in files where linaria is collocated with components.

## Reproducible Demo

add linaria no an empty storybook project https://github.com/storybookjs/sandboxes/tree/next/preact-vite/default-ts/after-storybook

## Solution

Disable prefresh.

Pseudo code
```tsx
const flatten = <T, >(setof: ReadonlyArray<T | T[]>): T[] => setof.flatMap(x => Array.isArray(x) ? x : [x])

export const viteFinal = ({plugins, ...rest}) => {
		return {
			plugins: 
                                // flatten plugins as preact-vite is an array - https://github.com/preactjs/preset-vite/blob/main/src/index.ts#L193
                                flatten<{name:string}>(plugins)
				// remove prefresh breaking linaria
				.filter(({name}) => name !== 'prefresh'),
			...rest
		}
	},

```

**Comment by mxeff** [comment]
I've encountered the same bug in my project. It's also reproducible in this StackBlitz by changing the boolean `prefreshEnabled` in `vite.config.ts` to `true` and refreshing the integrated browser a few times if needed:
https://stackblitz.com/edit/vitejs-vite-i6wmm8?file=vite.config.ts

My best guess is that this is some issue around this part in the `shaker-plugin.ts` - maybe because Prefresh is not a named and/or referenced import?
https://github.com/callstack/linaria/blob/master/packages/shaker/src/plugins/shaker-plugin.ts#L341

Anyway, as @theKashey mentioned, this is solved by disabling Prefresh in the Preact-Vite-Plugin (setting `prefreshEnabled` option to `false`) at the cost of developers QoL because a full refresh is required for updates.

**Comment by layershifter** [comment]
This have been fixed in #1343 and works with `@linaria/vite@5.0.3`: https://stackblitz.com/edit/vitejs-vite-11dfzy
